src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -411,6 +412,9 @@ private void updatePatronTransactionOnRequestChange(RequestDTO requestDTO, InnRe
     if (requestDTO.getStatus() == CLOSED_CANCELLED &&
       (transaction.getState() == PATRON_HOLD || transaction.getState() == TRANSFER)) {
       log.info("Updating patron hold transaction {} on cancellation of a request {}", transaction.getId(), requestDTO.getId());
+      if (transaction.getState() != ITEM_SHIPPED) {
+        clearPatronTransactionAndItemRecord(requestDTO.getItemId(), transaction);
```

Comment:
It's better to place patron info removing part at the very end of this block, after `notifier.reportCancelItemHold(transaction);`

Reply:
Done